synapse/util/caches/lrucache.py

```diff
@@ -111,12 +130,15 @@ def delete_node(node):
             prev_node.next_node = next_node
             next_node.prev_node = prev_node
 
+            deleted_len = 1
             if size_callback:
-                cached_cache_len[0] -= size_callback(node.value)
+                deleted_len = size_callback(node.value)
+                cached_cache_len[0] -= deleted_len
```

Comment:
I'm not sure we want to being doing eviction metrics based on size of the entries? I'd have thought we'd want evictions to be comparable to hit/misses which aren't based on the size of the entries?

Reply:
hum. I wanted to be able to compare it with the "cache size" graph, which is of course measured in terms of the size of the entries.

One of the problems we have at the moment is that a single room can monopolise the cache, which means that a single eviction can empty the entire cache. I wanted to highlight this happening - hence basing it on entry size rather than treating all evictions as identical.

Of course, we could compromise and monitor both...

Reply:
Fair enough, make sense. I wander if we should call it something other than "evictions" to make it clear? Maybe "evicted_size" or something?